frontend/src/app/main/ui/workspace/context_menu.cljs

```diff
@@ -33,8 +34,33 @@
   (dom/prevent-default event)
   (dom/stop-propagation event))
 
+
+(mf/defc element-icon
+  [{:keys [shape] :as props}]
+  (case (:type shape)
+    :frame i/artboard
+    :image i/image
+    :line i/line
+    :circle i/circle
+    :path i/curve
+    :rect i/box
+    :text i/text
+    :group (if (some? (:component-id shape))
+             i/component
+             (if (:masked-group? shape)
+               i/mask
+               i/folder))
+    :bool (case (:bool-type shape)
+            :difference   i/bool-difference
+            :exclude      i/bool-exclude
+            :intersection i/bool-intersection
+            #_:default    i/bool-union)
+    :svg-raw i/file-svg
+    nil))
+
+
 (mf/defc menu-entry
-  [{:keys [title shortcut on-click children] :as props}]
+  [{:keys [title shortcut on-click children selected has-icon? shape] :as props}]
```

Comment:
`selected` should be `selected?`

Reply:
Not sure if it's possible, but I'll prefer than, instead of passing a shape and calculating here what is the icon, to pass directly the icon, and do the calculation outside. So this component does not need to know anything about shapes.

Reply:
The has-icon? naming was really bad, I change it and move the icon check to its parent as you suggest. Tell me if it's not clear enough yet.